src/components/BadgeOverlay/styled/index.ts

```diff
@@ -0,0 +1,80 @@
+import styled from 'styled-components/native';
+
+import { BaseProps } from '../../types';
+import { variantOverlay, positionOverlay, overlapOverlay } from '../types';
+import {
+    COLOR_ERROR,
+    COLOR_WHITE,
+    COLOR_BRAND,
+    COLOR_SUCCESS,
+    COLOR_WARNING,
+} from '../../../styles/colors';
+
+const variants = {
+    error: COLOR_ERROR,
+    brand: COLOR_BRAND,
+    warning: COLOR_WARNING,
+    success: COLOR_SUCCESS,
+};
+
+const positionsRectangle = {
+    'top-left': 'top:0;left:0;transform: scale(1) translate(-10px,-10px)',
+    'top-right': 'top:0;right:0;transform: scale(1) translate(10px,-10px)',
+    'bottom-left': 'bottom:0;left:0;transform: scale(1) translate(-10px,10px)',
+    'bottom-right': 'bottom:0;right:0;transform: scale(1) translate(10px,10px)',
+};
+
+const positionsCircle = {
+    'top-left': 'top:11;left:11;transform: scale(1) translate(-10px,-10px)',
+    'top-right': 'top:11;right:11;transform: scale(1) translate(10px,-10px)',
+    'bottom-left':
+        'bottom:11;left:11;transform: scale(1) translate(-10px,10px)',
+    'bottom-right':
+        'bottom:11;right:11;transform: scale(1) translate(10px,10px)',
+};
+
+const positions = {
+    rectangle: positionsRectangle,
+    circle: positionsCircle,
+};
+
+const overlap = {
+    rectangle: 'border-radius: 0',
+    circle: 'border-radius: 999',
+};
+
+interface IOverlay extends BaseProps {
+    overlap: overlapOverlay;
+}
+
+export const StyledOverlayContainer = styled.View<IOverlay>`
+    position: relative;
+    ${props => overlap[props.overlap]}
```

Comment:
the component should fallback nicely if the user pass `overlap="wrong-value"` so we need to set a default value:
```${props => overlap[props.overlap] || 'border-radius: 999'}```

Reply:
as fallback use a value of the same object, so as not to lose consistency

```jsx
  ${props => overlap[props.overlap] || overlap.circle }
```